app/controllers/camaleon_cms/admin/media_controller.rb

```diff
@@ -71,13 +62,13 @@ def actions
           cama_uploader.delete_file(params[:folder].gsub('//', '/'))
           render plain: ''
         when 'crop_url'
-          unless params[:url].start_with?('data:')
-            params[:url] = (params[:url].start_with?('http') ? '' : current_site.the_url(locale: nil)) + params[:url]
-          end
-          r = if local_url?(params[:url])
-                { error: t('camaleon_cms.admin.media.local_upload_denied') }
+          user_url = params[:url].to_s
```

Comment:
Instead of receiving the file path (local files, the controller should expect for the file ID: `params[:file_id]`)

Reply:
What do you mean, @owen2345 ? The controller is receiving here the `url` param - `crop_url` action on uploading files from URLs.

Reply:
I assume that local crops is coming from media_manager.js#91 `return $.fn.upload_url({ url: data.url, name: cropName.join('.') })` where `data` holds the file information, then change it to pass the media.id instead of the URL (that could remove all the possible insecurities for local files)

Reply:
What do you think @texpert @brian-kephart

Reply:
I don't fully understand what's going on here. Using the old line numbers, it looks to me like L75 can in certain circumstances change `params[:url]` to an empty string, then that empty string is passed to `cama_tmp_upload` on L80. I think that these changes removed that possibility. Is that what you're referring to @owen2345?

If that is a necessary possibility, then I think it should be refactored to be more understandable by extracting some methods and not mutating the `params`.

Reply:
Never mind. You already fixed!